server/auth/login.go

```diff
@@ -0,0 +1,59 @@
+package auth
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+
+	"github.com/8tomat8/SSU-Golang-252-Chat/database"
+	"github.com/8tomat8/SSU-Golang-252-Chat/loger"
+	"github.com/jinzhu/gorm"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func ByName(UserName string) (*User, error) {
+	db, err := database.GetStorage()
+	if err != nil {
+		loger.Log.Errorf("Failed to open db", err)
+		return nil, err
+	}
+	res, err := byQuery(db.Where("username=?", UserName))
+	if err != nil {
+		loger.Log.Errorf("Search in db failed")
+		return nil, err
+	}
+	return res, err
+}
+
+func Login(username, password string) error {
```

Comment:
I am with Volodymyr decided to write token into client connection, and appropriate every time user log on